src/fa-screenshot.sh

```diff
@@ -0,0 +1,182 @@
+#!/bin/bash
+
+# Script to take screenshots in the correct format
+
+# how to run : fa-screenshot fileName tp
+# filename (optional)- name of the file to save screenshot to. extension png is automatically added.
+# tp (optional)- use tinypng instead of pngquant for compression
+#TODO: switch for accepting window positioning and sizing parameters.
+
+
+# Author - Sanjay
+
+
+function printFileAttribs() {
+  attributes=`/usr/local/bin/identify ${absFile}`
+  printOut "Current dimensions of the screenshot are : "`echo ${attributes} | cut -d' ' -f3`
+  printOut "Current size of the screenshot is : "`echo ${attributes} | cut -d' ' -f7`
+}
+
+function compressUsingTPNG() {
+  echo "-- Using TinyPNG for compression"
+
+  API_KEY=`printenv | grep TINYPNG_API_KEY | awk -F '=' '{print $2}'`
+  if [ "${API_KEY}" == "" ]; then
+    echo "Error : TinyPNG API key not found in environment. Not compressing".
+  else
+    compressedFile=`curl -s --user api:${API_KEY} --data-binary @${absFile} -i https://api.tinify.com/shrink | grep -i location | awk '{print $2}' | sed 's/.$//'`
+    printOut "-- Downloading from TinyPNG : ${compressedFile}"
+    curl -s -X GET "${compressedFile}" --output ${absFile}
+  fi
+}
+
+function printOut() {
+  if [ ${silent} == "no" ]; then
+    echo
+    echo ${1}
+  fi
+}
+
+
+function usage() {
+  #print pretty usage and exit
+  cat <<HELP_USAGE
+
+    $0  [-s] [-t] [-d] <destination folder> [-h]
+
+   -s  Silent mode
+   -t  Use TinyPNG API instead of pngquant library
+   -d  Move screenshots to given folder
+   -h  Print this usage
+HELP_USAGE
+  exit 0;
+}
+
+
+silent="no"
+useTP="no"
+destination=""
+while getopts ":std:h" options; do
+    case "${options}" in
+        s)
+            silent="yes"
+            ;;
+        t)
+            useTP="yes"
+            ;;
+        d)
+           destination=${OPTARG}
+           if [[ "${destination}" == "" || ! -d ${destination} ]]; then
+             echo "Please specify valid destination folder"
+             exit 1;
+           fi
+           ;;
+         h)
+           usage
+           ;;
+        *)
+            usage
+            ;;
+    esac
+done
+shift $((OPTIND-1))
+
+
+printOut "-- checking if imagemagick is installed..."
+if [ ! -f "/usr/local/bin/convert" ]; then
+  echo "Installing imagemagick..."
+  brew install imagemagick;
+else
+  printOut "imagemagick is installed"
+fi
+
+printOut "-- Checking whether pngquant is installed"
+if [ ! -f "/usr/local/bin/pngquant" ]; then
+  echo "Installing pngquant"
+  brew install pngquant
+else
+  printOut "pngquant is installed"
+fi
+
+
+printOut "-- Checking whether safari is running"
+safaricount=`ps cax | grep -i "safari.app" | grep -iv grep | wc -l`;
+
+if [ "${safaricount}" == "0" ]; then
+  echo "Safari is not running. Exiting..."
+  exit 1;
+else
+  printOut "Safari is running"
+fi
+
+
+printOut "-- Resizing safari window..."
+osascript <<EOD
+set theApp to "Safari"
```

Comment:
Optionally could tell the system to change to light mode before taking the screenshot

```
tell application "System Events"
    tell appearance preferences
        set dark mode to false
    end tell
end tell
```